os/mm/kmm_heap/kmm_memalign.c

```diff
@@ -97,18 +97,22 @@ void *kmm_memalign_at(int heap_index, size_t alignment, size_t size)
 		return NULL;
 	}
 
+	if (size == 0) {
+		return NULL;
+	}
+
 	kheap = kmm_get_heap();
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	ARCH_GET_RET_ADDRESS
 	ret = mm_memalign(&kheap[heap_index], alignment, size, retaddr);
 	if (ret == NULL) {
-		mm_manage_alloc_fail(&kheap[heap_index], 1, size, KERNEL_HEAP);
+		mm_manage_alloc_fail(&kheap[heap_index], heap_index, heap_index + 1, size, KERNEL_HEAP);
```

Comment:
Logically, end index should be `heap_index`. But you implemented it with `<`. The end is set as `heap_index + 1`. I think you should modify the function to compare with `<=` for end and you should give the same value for the start and the end.
```
mm_manage_alloc_fail(xxx, heap_index, heap_index, size. yyy);

void mm_manage_alloc_fail(xxx, int start_idx, int end_idx, size, yyy)
{
    for (i = start_idx; i<= end_idx; i++) {
    ...
   }
}
```

Reply:
I want to check just `heap[heap_index]`. So I think start can be `heap_index`, end can be `heap_index + 1` and compare the end with `<`.
If modified to `<=`, we should pass the end param with `-1`.
For example, 
````
mm_manage_alloc_fail(kheap, 0, CONFIG_KMM_NHEAPS, size, KERNEL_HEAP);
```
In this case, we should pass `CONFIG_KMM_NHEAPS - 1`, instead of `CONFIG_KMM_NHEAPS`.

Reply:
In my opinion, a function should have certain operation and it should be used easily without code analysis. When I see end index, it should be `heap_index` or `CONFIG KMM NHEAPS - 1`. When we see the code, then we know +1 due to <.

Reply:
Changed. Thanks.